documentation/src/main/asciidoc/mapper-orm-bridge.asciidoc

```diff
@@ -0,0 +1,1353 @@
+* one method to convert the document identifier back to the original property value.
+
+Below is an example of a custom identifier bridge that converts
+a custom `BookId` type to its string representation and back:
+
+.Implementing and using an `IdentifierBridge`
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/mapper/orm/bridge/identifierbridge/simple/BookIdBridge.java[tags=include]
+----
+<1> The bridge must implement the `IdentifierBridge` interface.
+One generic parameters must be provided:
+the type of property values (values in the entity model).
+<2> The `toDocumentIdentifier` method takes the property value and a context object as parameters,
+and is expected to return the corresponding document identifier.
+It is called when indexing,
+but also when parameters to the search DSL
+<<search-dsl-argument-type,must be transformed>>,
+in particular for the <<search-dsl-predicate-id,ID predicate>>.
+<3> The `fromDocumentIdentifier` methods takes the document identifier and a context object as parameters,
+and is expected to return the original property value.
+It is called when mapping search hits to the corresponding entity.
+
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/mapper/orm/bridge/identifierbridge/simple/Book.java[tags=include;!getters-setters]
+----
+<1> Map the property to the document identifier.
+<2> Instruct Hibernate Search to use our custom identifier bridge.
+It is also possible to reference the bridge by its name, in the case of a CDI/Spring bean.
+====
+
+=== Type resolution
+
+By default, the identifier bridge's property type is determined automatically,
+using reflection to extract the generic type argument of the `Identifier` interface.
+
+For example, in `public class MyBridge implements IdentifierBridge<BookId>`,
+the property type is resolved to `BookId`:
+the bridge will be applied to properties of type `BookId`.
+
+The fact that the type is resolved automatically using reflection brings a few limitations.
+In particular, it means the generic type argument cannot be just anything;
+as a general rule, you should stick to literal types (`MyBridge implements IdentifierBridge<BookId>`)
+and avoid generic type parameters and wildcards
+(`MyBridge<T extends Number> implements IdentifierBridge<T>`,
+`MyBridge implements IdentifierBridge<List<? extends Number>>).
+
+If you need more complex types,
+you can bypass the automatic resolution and specify types explicitly
+using an <<mapper-orm-bridge-identifierbridge-identifierbinder,`IdentifierBinder`>>.
+
+=== Compatibility across indexes with `isCompatibleWith()`
+
+An identifier bridge is involved in indexing,
+but also in the search DSLs,
+to convert values passed to the <<search-dsl-predicate-id,`id` predicate>>
+to a document identifier that the backend will understand.
+
+When creating an `id` predicate targeting multiple indexes,
```

Comment:
We're describing the mapping here. So maybe we can say 
s/targeting multiple indexes/targeting multiple entities?